frontend/projects/upgrade/src/app/shared/services/common-text-helpers.service.ts

```diff
@@ -0,0 +1,56 @@
+import { Injectable } from '@angular/core';
+
+/**
+ * A service class that provides common text formatting helpers.
+ *
+ * This class contains static methods to format group types, convert names to keys, etc.
+ * The methods are static so they can be used in pure functions like selectors without
+ * needing to create an instance of the service.
+ */
+@Injectable()
+export class CommonTextHelpersService {
+  /**
+   * Formats an array of group types into an array of objects with value and viewValue properties.
+   *
+   * @param groupTypes - An array of group type strings, as defined in context-metadata.
+   * @returns An array of objects where each object contains a value and a viewValue property, or [] if groupTypes is falsy.
+   *
+   * The method is static so it can be used in pure functions like selectors without needing
+   * to create an instance of the service.
+   */
+  public static formatGroupTypes(groupTypes: string[]): { value: string; viewValue: string }[] {
+    return (
+      groupTypes?.map((groupType) => {
+        return { value: groupType, viewValue: CommonTextHelpersService.formatGroupTypeViewValue(groupType) };
+      }) ?? []
```

Comment:
shouldn't we use || instead ?? here.

Reply:
meh I think that this is trying to be too clever in the first place and something like this is probably clearest. Really I'm just trying to return an empty array if groupTypes is not an array.

```
public static formatGroupTypes(groupTypes: string[]): { value: string; viewValue: string }[] {
   if (!Array.isArray(groupTypes)) {
      return [];
    }
    return groupTypes.map((groupType) => {
      return { value: groupType, viewValue: CommonTextHelpersService.formatGroupTypeViewValue(groupType) };
    });
  }
```

Reply:
can you replace the current code with the one you added here, it might be easy to understand.

Reply:
it is, this comment is outdated now